internal/distributed/proxy/httpserver/utils.go

```diff
@@ -1748,6 +1748,17 @@ func RequestHandlerFunc(c *gin.Context) {
 	c.Writer.Header().Set("Access-Control-Allow-Credentials", "true")
 	c.Writer.Header().Set("Access-Control-Allow-Headers", "Content-Type, Content-Length, Accept-Encoding, X-CSRF-Token, Authorization, accept, origin, Cache-Control, X-Requested-With")
 	c.Writer.Header().Set("Access-Control-Allow-Methods", "GET, HEAD, POST, PUT, DELETE, OPTIONS, PATCH")
+	c.Writer.Header().Set("X-Content-Type-Options", "nosniff") // Prevents MIME sniffing
+
+	maxAge := paramtable.Get().HTTPCfg.HSTSMaxAge.GetValue()
+	includeSubDomains := paramtable.Get().HTTPCfg.HSTSIncludeSubDomains.GetAsBool()
+
+	hstsValue := fmt.Sprintf("max-age=%s", maxAge)
+	if includeSubDomains {
+		hstsValue += "; includeSubDomains"
+	}
+	c.Writer.Header().Set("Strict-Transport-Security", hstsValue)
```

Comment:
Let's make this modification atomic. how about change to this:

       enableHSTS := paramtable.Get().HTTPCfg.EnableHSTS.GetAsBool()
       if enableHSTS {
                maxAge := paramtable.Get().HTTPCfg.HSTSMaxAge.GetValue()
                hstsValue := fmt.Sprintf("max-age=%s", maxAge)
                includeSubDomains := paramtable.Get().HTTPCfg.HSTSIncludeSubDomains.GetAsBool()
                if includeSubDomains {
                    hstsValue += "; includeSubDomains"
                }
                c.Writer.Header().Set("Strict-Transport-Security", hstsValue)
	}